handlers/utils/compute_utils.go

```diff
@@ -0,0 +1,1186 @@
+	if IsNoOp(instance.Data) {
+		return true
+	}
+
+	client := docker.GetClient(DefaultVersion)
+	container := GetContainer(client, instance, false)
+	return isStopped(client, container)
+}
+
+func DoInstanceForceStop(request *model.InstanceForceStop) error {
+	client := docker.GetClient(DefaultVersion)
+	time := time.Duration(0)
+	stopErr := client.ContainerStop(context.Background(), request.ID, &time)
+	if stopErr != nil {
+		logrus.Error(stopErr)
+		return stopErr
+	}
+	logrus.Infof("container id %v is forced to be stopped", request.ID)
+	return nil
+}
+
+func DoInstanceInspect(inspect *model.InstanceInspect) (types.ContainerJSON, error) {
+	client := docker.GetClient(DefaultVersion)
+	containerID := inspect.ID
+	if containerID == "" {
+		return types.ContainerJSON{}, fmt.Errorf("container with id [%v] not found", containerID)
+	}
+	containerList, _ := client.ContainerList(context.Background(), types.ContainerListOptions{All: true})
+	result := findFirst(containerList, func(c *types.Container) bool {
+		return idFilter(containerID, c)
+	})
+	if result == nil {
+		name := fmt.Sprintf("/%s", inspect.Name)
+		result = findFirst(containerList, func(c *types.Container) bool {
+			return nameFilter(name, c)
+		})
+	}
+	if result != nil {
+		logrus.Infof("start inspecting container with id [%s]", result.ID)
+		inspectResp, err := client.ContainerInspect(context.Background(), result.ID)
+		if err != nil {
+			logrus.Error(err)
+			return types.ContainerJSON{}, err
+		}
+		logrus.Infof("container with id [%s] inspected", result.ID)
+		return inspectResp, nil
+	}
+	return types.ContainerJSON{}, fmt.Errorf("container with id [%v] not found", containerID)
+}
+
+func nameFilter(name string, container *types.Container) bool {
+	names := container.Names
+	if names == nil || len(names) == 0 {
+		return false
+	}
+	found := false
+	for _, n := range names {
+		if strings.HasSuffix(n, name) {
+			found = true
+			break
+		}
+	}
+	return found
+}
+
+func IsInstanceRemoved(instance *model.Instance) bool {
+	client := docker.GetClient(DefaultVersion)
+	container := GetContainer(client, instance, false)
+	return container == nil
+}
+
+func DoInstanceRemove(instance *model.Instance, progress *progress.Progress) error {
+	client := docker.GetClient(DefaultVersion)
+	container := GetContainer(client, instance, false)
+	if container == nil {
+		return errors.New("container not found")
+	}
+	return removeContainer(client, container.ID)
+}
+
+func PurgeState(instance *model.Instance) {
+	client := docker.GetClient(DefaultVersion)
+	container := GetContainer(client, instance, false)
+	if container == nil {
+		return
+	}
+	dockerID := container.ID
+	contDir := ContainerStateDir()
+	files := []string{path.Join(contDir, "tmp-"+dockerID), path.Join(contDir, dockerID)}
+	for _, f := range files {
+		if _, err := os.Stat(f); err == nil {
+			os.Remove(f)
+		}
+	}
+}
+
+func getInstanceHostMapData(event *revents.Event) map[string]interface{} {
```

Comment:
change signature to also return an error and return an error where you are ignoring or swallowing them